File: src/DotNetPad/DotNetPad.Applications/Samples/NameOfOperator.cs
namespace Waf.DotNetPad.Samples;

public static class NameOfOperator
{
    private static readonly ObservableCollection<string> list = new();

    public static void Main()
    {
        // Use nameof to compare with the property name provided by the event args.
        ((INotifyPropertyChanged)list).PropertyChanged += ListPropertyChanged;
        list.Add("Luke");
        list.Add("Han");

        // Use nameof with the ArgumentNullException constructor.
        try
        {
            ArgumentNullCheck(null);
        }
        catch (ArgumentNullException ex)
        {
            Console.WriteLine("Expected exception was thrown: " + ex.Message);
        }
    }

    private static void ListPropertyChanged(object? sender, PropertyChangedEventArgs e)
    {
        if (e.PropertyName == nameof(list.Count))
        {
            Console.WriteLine("Count: " + list.Count);
        }
    }

    private static void ArgumentNullCheck(IReadOnlyList<string>? list)
    {
        if (list is null) throw new ArgumentNullException(nameof(list));
    }
}

File: src/DotNetPad/DotNetPad.Applications/Samples/NameOfOperator.vb
Imports System
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.ComponentModel

Namespace Waf.DotNetPad.Samples
    Module NameOfOperator
        Private list As ObservableCollection(Of String)

        Sub Main()
            ' Use nameof to compare with the property name provided by the event args.
            list = New ObservableCollection(Of String)
            AddHandler CType(list, INotifyPropertyChanged).PropertyChanged, AddressOf ListPropertyChanged
            list.Add("Luke")
            list.Add("Han")

            ' Use nameof with the ArgumentNullException constructor.
            ArgumentNullCheck(Nothing)
        End Sub

        Sub ListPropertyChanged(sender As Object, e As PropertyChangedEventArgs)
            If (e.PropertyName = nameof(list.Count)) Then
                Console.WriteLine("Count: " + list.Count.ToString)
            End If
        End Sub

        Sub ArgumentNullCheck(list As IReadOnlyList(Of String))
            If (list Is Nothing) Then
                Throw New ArgumentNullException(nameof(list))
            End If
        End Sub
    End Module
End Namespace